### DISCUSSION TITLE
Two questions

### DISCUSSION THREAD

**Discussion by realchandan** [Malinskiy/adam#110] [Q&A] [answered]
Hi there,

I read your medium article that explained about how adb's internals work and the issues you had with ddmlib that led to the creation of this library.

I have two questions regarding this library. I hope you don't mind answering them.

I'm wondering whether your library connects to adb daemon running on some phone, through a middle entity (adb server) or it directly connects to the adb daemon on another phone.

Also, I genuinely don't understand how your library differs from [adblib](https://github.com/tananaev/adblib).

Thanks

**Comment by Malinskiy** [comment]
Hi,

1. This project currently has a client implementation that communicates with adb server as well as emulator-specific transport
2. Which library are you referring to by `your` library [out of 8 of them](https://search.maven.org/search?q=g:com.malinskiy.adam)?

Cheers

**Reply by realchandan** [reply]
Hi again,

1. I totally missed that it's not an android library, so If JetBrains desires they can swap current ddmlib with adam for their products :)
2. I was referring to [this](https://central.sonatype.com/artifact/com.malinskiy.adam/adam) but I got my answer now, libadb connects directly to adb daemon of a phone.

Thanks

**Reply by Malinskiy** [reply, marked as answer]
1. I tried to swap ddmlib in Android Studio a couple of years ago and unfortunately it's [write-only open source](https://x.com/anton_malinskiy/status/1582395455505596417?s=20) so unless someone at JB wants to spend time doing it it's unlikely to happen. But I suspect swapping ddmlib out will improve the global warming situation by not warming up developer machines so much
2. :+1: you can also take a look at https://github.com/mobile-dev-inc/dadb since they also have a use-case of connecting directly to the android device

**Reply by realchandan** [reply]
Hi,

Thanks for the info, got what I needed!

Cheers
